### PR TITLE
Daniel rusu

### PR DESCRIPTION
**Pull request by RDaniel99**
https://trello.com/c/shTmJ3N7/10-email-system

### CODE REVIEW

File: controllers/authController.js [hunk 1]
```diff
@@ -69,3 +72,27 @@ exports.register = async (req, res) => {
     })
   }
 }
+
+async function sendVerifMail(mail) {
```
Comment (procateodor) [inline controllers/authController.js]: Let's put this function in a separate file under `utils` folder.

File: controllers/authController.js [hunk 2]
```diff
@@ -48,6 +49,8 @@ exports.register = async (req, res) => {
         message: 'User already exists!'
       })
     }
+    
+    sendVerifMail(req.body.email).catch(console.error);
```
Comment (procateodor) [inline controllers/authController.js]: We have to send the email right before to respond back to the client.

File: controllers/authController.js [hunk 3]
```diff
@@ -69,3 +72,27 @@ exports.register = async (req, res) => {
     })
   }
 }
+
+async function sendVerifMail(mail) {
+    let transporter = nodemailer.createTransport({
+      service: "Gmail",
+      auth: {
+        user: 'testdsc242@gmail.com',
```
Comment (procateodor) [inline controllers/authController.js]: We should put this configs in out config files, not directly here.

File: controllers/authController.js (hunk 2 above)
Comment (procateodor) [reply controllers/authController.js]: Also, we don't send a verify email, just a welcome email.
So the function should be call `sendEmail`.

File: models/user.js [hunk 4]
```diff
@@ -6,6 +6,12 @@ const userSchema = new Schema({
   },
   password: {
     type: String
+  },
+  isMailVerified: {
```
Comment (procateodor) [inline models/user.js]: don't change the user schema.
all users all verified by default.
we will handle this in the future.

File: controllers/authController.js [hunk 5]
```diff
@@ -69,3 +72,27 @@ exports.register = async (req, res) => {
...
+        pass: 'testdsc0'
+      }
+    });
+
+    let mailOption = {
+      from: 'Test Test <testdsc242@gmail.com>', // sender address
+      to: 'rusudaniel5799@gmail.com', // list of receivers
+      subject: "Hello", // Subject line
```
Comment (procateodor) [inline controllers/authController.js]: let's add in our config file the subject and from.
`to` should be the email from the user.
please add the template with a replace variable ability.

**Review by procateodor — CHANGES_REQUESTED** [review]
Just a few comments.

File: controllers/authController.js (hunk 5 above)
Comment (RDaniel99) [reply controllers/authController.js]: How can I use template1.html as variable to mail body?

Comment (procateodor) [reply controllers/authController.js]: @RDaniel99 citesti templatet-ul si dupa inlocuiesti ce e de inlocuit si dupa il pui la html

**Comment by RDaniel99** [comment]
[Email system.](https://trello.com/c/shTmJ3N7/10-email-system)

**Review by procateodor — APPROVED** [review]

**Pull request merged**